src/main/java/hudson/tasks/test/TestResult.java

```diff
@@ -276,19 +286,22 @@ public String toPrettyString() {
     }
 
     /**
-     * Annotate some text -- what does this do? 
+     * Annotate text by turning it into a clickable link and then sanitizing it using owasp-java-html-sanitizer
+     * @see <a href="https://www.javadoc.io/doc/com.googlecode.owasp-java-html-sanitizer/owasp-java-html-sanitizer/20160628.1/org/owasp/html/HtmlPolicyBuilder.html">owasp HtmlPolicyBuilder javadoc</a>
      * @param text Text to use to annotate the actions.
      *
      * @return the provided text HTML-escaped.
      */
     public String annotate(String text) {
         if (text == null)
                 return null;
-        text = text.replace("&", "&amp;").replace("<", "&lt;");
+
+        text = text.replaceAll("\\b(https?://[^\\s)>]+)", "<a href=\"$1\">$1</a>");
```

Comment:
text is essentially user supplied.

have you thought about if it contained non allowed HTML ( e.g. `this is <strong>some text</strong> in an assertion that failed`   
what happens in this case - does the message loose fidelity by being interpretted as pseudo HTML?
(is it displayed as the litteral "this is <strong>some text</strong> in an assertion that failed" or the litteral "this is some text in an assertion that failed" or something else?

I would semi expect the sanitizer to throw away the `strong` part - but what it really wanted is to see the underlying raw text here. (that is I expect to see in the web UI the `<strong>` part.   but I have a feeling this will be thrown away.

Reply:
FWIW that screenshot shows you get different output in different places - in on the error message has "strong" and in the test overview page it does not...    😱